File: 11137 Ingenuous Cubrency/solution.c
#include <stdio.h>
#define MAX_AMOUNT 9999

long dp[MAX_AMOUNT + 1] = {1};
int coins[] = {
    1, 8, 27, 64, 125, 216, 343, 512, 729, 1000, 
    1331, 1728, 2197, 2744, 3375, 4096, 4913, 5832, 6859, 8000, 
    9261
};

int main() {
    for(int i = 0; i < 21; ++i) {
        for(int j = coins[i]; j <= MAX_AMOUNT; ++j)
            dp[j] += dp[j - coins[i]];
    }
    int n;
    while(scanf("%d", &n) == 1)
        printf("%ld\n", dp[n]);
	return 0;
}

File: 11137 Ingenuous Cubrency/solution.java
import java.io.*;

public class Main {
    static int MAX_AMOUNT = 9999;
    static long[] dp = new long[MAX_AMOUNT + 1];
    static int[] coins = {
        1, 8, 27, 64, 125, 216, 343, 512, 729, 1000, 
        1331, 1728, 2197, 2744, 3375, 4096, 4913, 5832, 6859, 8000, 
        9261
    };
	public static void main(String[] args) throws IOException {
		dp[0] = 1;
		for(Integer coin : coins) {
		    for(int i = coin; i <= MAX_AMOUNT; ++i)
		        dp[i] += dp[i - coin];
		}
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
		StringBuilder output = new StringBuilder();
		while(st.nextToken() == StreamTokenizer.TT_NUMBER)
		    output.append(dp[(int)st.nval]).append('\n');
		System.out.print(output);
	}
}
